src/aiida/calculations/stash.py

```diff
@@ -69,6 +72,12 @@ def define(cls, spec):
         }
 
     def prepare_for_submission(self, folder):
+        if self.inputs.source_node.computer.uuid != self.inputs.metadata.computer.uuid:
+            EXEC_LOGGER.warning(
+                'The computer of the source node and the computer of the calculation must be the same.'
+                ' THIS MIGHT RESULT IN A SILENT FAILURE!'
```

Comment:
Can you explain maybe to the user, when one would like to use two different computers.